internal/repository/advancer.go

```diff
@@ -0,0 +1,196 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package repository
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	. "github.com/cartesi/rollups-node/internal/node/model"
+	"github.com/cartesi/rollups-node/internal/node/nodemachine"
+	"github.com/jackc/pgx/v5"
+)
+
+var ErrAdvancerRepository = errors.New("advancer repository error")
+
+type AdvancerRepository struct{ *Database }
+
+func (repository *AdvancerRepository) GetInputs(
+	ctx context.Context,
+	appAddresses []Address,
+) (map[Address][]*Input, error) {
+	result := map[Address][]*Input{}
```

Comment:
```suggestion
	result := map[Address][]Input{}
```

Reply:
It is indeed a slice o pointers (to Inputs). Not a slice of Inputs.